vsphere/datadog_checks/vsphere/constants.py

```diff
@@ -40,7 +40,7 @@
 DEFAULT_BATCH_COLLECTOR_SIZE = 500
 DEFAULT_METRICS_PER_QUERY = 500
 UNLIMITED_HIST_METRICS_PER_QUERY = float('inf')
-DEFAULT_MAX_QUERY_METRICS = 256
+DEFAULT_MAX_QUERY_METRICS = 256.0
```

Comment:
int should work better

Reply:
That's because it's later compared to `UNLIMITED_HIST_METRICS_PER_QUERY = float('inf')` which is float.

Reply:
I really question the benefit/cost ratio of type hints in cases like this, and they happen often...

Reply:
Agree. We shouldn't be too strict on cases like this and just use a simple solution.